tests/queries/0_stateless/03261_secondary_indicies_by_default.sql

```diff
@@ -0,0 +1,85 @@
+set mutations_sync = 2;
+
+DROP TABLE IF EXISTS tbl;
+DROP TABLE IF EXISTS tbl2;
+
+CREATE TABLE tbl
+(
+    key Int,
+    x Int,
+    y Int,
+    INDEX x_idx x TYPE minmax GRANULARITY 1
+)
+Engine=MergeTree()
+ORDER BY key;
+
+SELECT 'Default';
+SHOW INDICES FROM tbl;
+
+SELECT 'Enable enable_minmax_index_for_all_numeric_columns';
+ALTER TABLE tbl MODIFY SETTING enable_minmax_index_for_all_numeric_columns = true;
```

Comment:
Thoughts:
- According to the .reference file, l. 20 actually creates min-max indices on the table columns. As a user, I would be surprised. Maybe I am wrong here but one would expect that changing a setting is a very cheap operation. An alternative approach is to forbid modifying the setting (i.e. throw an exception) after the table has been created. The only way to provide a non-default setting would then be at table creation time. ALso, since the two new settings are merge tree settings, it would also be interesting to understand what happens if other merge tree settings are changed. Is there ever the case that a setting change can inflict major costs?

- Perhaps the operation is still cheap in the sense that the new index is not materialized for existing parts. In my view, such a behavior is still questionable for two reasons. 1. Creating indices would be expected for a dedicated `CREATE INDEX` DDL statement but not for `MODIFY SETTINGS` DDL. 2. The entire use case of auto-creating min-max indices on existing tables is debatable (imho). In my view, it would be just fine to allow auto-min-max indices only for new tables. At least, this would de-risk the whole feature and users who like to create min-max indices on existing tables can very easily do so through existing DDL syntax.

In any case, it would be good if the test could include a check if the index is materialized or not (you will need to insert a few rows for that to test).

Reply:
Indeed this will not be a cheap operation to change this setting. 

>  Is there ever the case that a setting change can inflict major costs?
I don't think there are setting that would cause such major cost, atleast they don't cause metadata change.

> At least, this would de-risk the whole feature and users who like to create min-max indices on existing tables can very easily do so through existing DDL syntax.
Yes sounds good, we can allow this to be set only during creation, not sure if we do that for other settings. @alesapin what do you think?

Reply:
> In any case, it would be good if the test could include a check if the index is materialized or not (you will need to insert a few rows for that to test).

You mean does it get materialized after setting is modified ? If so, yes it does get materialized on new inserts & merges (i.e. same as ADD INDEX). Any suggestions on how to check this in test ?

Reply:
> Do you mean does it get materialized after setting is modified?

Yes.

> If so, yes it does get materialized on new inserts & merges (i.e. same as ADD INDEX). Any suggestions on how to check this in test ?

Good.

To test, please see [this example](https://fiddle.clickhouse.com/e7281ebc-e3d6-44fb-b8df-5d4f940cb4f6).

Reply:
Updated setting to be immutable and added tests to check if index is materialized after ADD COLUMN when settings are enabeled.